assets/js/components/ExecutionResults/CheckResultOutline.jsx

```diff
@@ -8,21 +8,29 @@ const TARGET_CLUSTER = 'TARGET_CLUSTER';
 const isExpect = ({ type }) => type === EXPECT;
 const isExpectSame = ({ type }) => type === EXPECT_SAME;
 
-function CheckResultOutline({
-  checkID,
-  agentsCheckResults,
-  expectationResults,
-  clusterName,
-}) {
-  const expectExpectationsCount = expectationResults.filter(isExpect).length;
-  const expectResults = agentsCheckResults.map((agentCheckResult) => {
+const expectExpectations = (expectationList) =>
+  expectationList.filter(isExpect);
+
+const getExpectExpectationsMet = (expectationEvaluations) =>
+  expectExpectations(expectationEvaluations).filter(
+    ({ return_value }) => return_value
+  );
+
+const extractExpectResults = (expectations, agentsCheckResults) => {
+  const expectExpectationsCount = expectExpectations(expectations).length;
+
+  if (expectExpectationsCount === 0) {
+    return [];
+  }
```

Comment:
You can have the array in place and if the length is 0 you can return it directly